pkg/services/grafana-apiserver/storage/entity/storage.go

```diff
@@ -241,17 +241,13 @@ func (s *Storage) GetList(ctx context.Context, key string, opts storage.ListOpti
 		}
 	}
 
-	// translate grafana.app/folder field selector to folder condition
-	fields := opts.Predicate.Field.Requirements()
-	for _, f := range fields {
-		if f.Field == "grafana.app/folder" {
-			if f.Operator != selection.Equals {
-				return apierrors.NewBadRequest("grafana.app/folder field selector only supports equality")
-			}
-
-			// select items in the spcified folder
-			req.Folder = f.Value
-		}
+	// translate grafana.app/folder field selector to the folder condition
+	fieldRequirements, err := ReadFieldRequirements(opts.Predicate.Field)
```

Comment:
this only works properly if we also remove that field from the predicate using Transform below, otherwise the predicate will apply it and filter everything out. It might make sense to change the signature of this function so it can also return an updated opts.Predicate.Field, or just have it accept opts.Predicate so it can perform the Transform.